var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -93,13 +93,19 @@ def setup_build_environment(self, env):
             env.set("PAPI_ROCM_ROOT", spec["hsa-rocr-dev"].prefix)
             env.set("HSA_TOOLS_LIB", "%s/librocprofiler64.so" % spec["rocprofiler-dev"].prefix.lib)
             env.append_flags("CFLAGS", "-I%s/rocprofiler/include" % spec["rocprofiler-dev"].prefix)
-            env.set(
-                "ROCP_METRICS", "%s/rocprofiler/lib/metrics.xml" % spec["rocprofiler-dev"].prefix
-            )
+            if os.path.exists("%s/rocprofiler/metrics.xml" % spec["rocprofiler-dev"].prefix.lib):
```

Comment:
Maybe simplify it with just this?
```suggestion
            env.set("ROCP_METRICS", find(spec["rocprofiler-dev"].prefix, "metrics.xml"))
```
I haven't yet tested this because the rocm stack takes so long to install.

Reply:
I only see string find() in python, which does not match the behavior described in your code. I could still write a `find_file` function that searches for a file inside a directory and returns it if found. However, this looks orthogonal to the build error. I think you are right. There might be a problem with PAPI master (specifically with the testing framework) that pops up with more recent versions of gcc. We had similar problems in the past when introducing testing with newer compiler versions.

Reply:
Another things that pops up is that your environment is using rocm-5.4.3, which is know to have some problem with the value of the counters reported. The latest known working version should be 5.3.0 but if you want to be safe I would go with 5.2.0 (which is the one I am using for my testing).

Reply:
This is not really my environment.  This is what spack is determining is the correct versions to use based on what was requested and the papi package definition.  If papi doesn't work with a version of rocm, then that should be included as a `conflict()` in the spack package.  Otherwise, users would not know what the problem is.  The default that spack builds with when you do a simple `spack install papi+rocm` should work without customization.

Reply:
Yes, you are right. I meant the environment in the test system you referenced.

Reply:
> If papi doesn't work with a version of rocm, then that should be included as a `conflict()` in the spack package. Otherwise, users would not know what the problem is. The default that spack builds with when you do a simple `spack install papi+rocm` should work without customization.

I can update the PR to include the conflicting rocm versions

Reply:
The `find()` is from spack in lib/spack/llnl/util/filesystem.py`.  I see it is used this way in other spack packages (e.g., openmpi).

> I only see string find() in python, which does not match the behavior described in your code. I could still write a `find_file` function that searches for a file inside a directory and returns it if found. However, this looks orthogonal to the build error.

Reply:
> I meant the environment in the test system you referenced.

It isn't being affected by the environment.  That is something that spack generally tries hard to avoid.  We should assume that spack will try to use the newest version of any package that it can if there are no other constraints.  This is why we need conflicts() if papi is known to not work with certain dependency versions.  In this case:

```
conflicts("^hip@5.4:")
```